session_0/item.rb

```diff
@@ -23,14 +23,20 @@ def initialize(name: '', price: 0, quantity: 1, category_id: 4,
   # Returns a boolean value whether than item is discounted i.e. the
   # discount deadline has been crossed or not.
   def discounted?
-    raise NotImplementedError # TODO
+    return Time.now > @discount_deadline ? false : true
   end
 
   # If the item is discounted, the current price is 
   # `price * (100 - discount rate) / 100`. Otherwise, it is same as the price.
   #
   # TODO: Implement instance method 'current_price'
-
+  def current_price
+    if discounted?
+      return @price * (100 - @discount_rate) /100
```

Comment:
You don't need to use `@price` as we have defined attribute accessor for price - using `price` works fine too. Likewise for other attributes.